sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/DefaultSynchronousMetricStorage.java

```diff
@@ -165,24 +158,39 @@ public MetricData collectAndReset(
       InstrumentationScopeInfo instrumentationScopeInfo,
       long startEpochNanos,
       long epochNanos) {
+    boolean reset = aggregationTemporality == AggregationTemporality.DELTA;
+
     // Grab accumulated measurements.
     Map<Attributes, T> accumulations = new HashMap<>();
     for (Map.Entry<Attributes, AggregatorHandle<T, U>> entry : activeCollectionStorage.entrySet()) {
-      boolean unmappedEntry = entry.getValue().tryUnmap();
-      if (unmappedEntry) {
-        // If able to unmap then remove the record from the current Map. This can race with the
-        // acquire but because we requested a specific value only one will succeed.
-        activeCollectionStorage.remove(entry.getKey(), entry.getValue());
+      if (reset) {
+        boolean unmappedEntry = entry.getValue().tryUnmap();
+        if (unmappedEntry) {
+          // If able to unmap then remove the record from the current Map. This can race with the
+          // acquire but because we requested a specific value only one will succeed.
+          activeCollectionStorage.remove(entry.getKey(), entry.getValue());
+        }
       }
-      T accumulation = entry.getValue().accumulateThenReset(entry.getKey());
+      T accumulation = entry.getValue().accumulateThenReset(entry.getKey(), reset);
```

Comment:
This code is pretty hard to understand and only exists to serve the `Bound{Instrument}` concept which has been shelved. If we don't see a realistic chance of the being incorporated into the spec, I suggest we rip it out and reap the benefits of reduced complexity.